x-pack/plugins/cases/public/components/create/case_owner_selection.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useCallback, useEffect } from 'react';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiIcon,
+  EuiKeyPadMenu,
+  EuiKeyPadMenuItem,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+
+import { euiStyled } from '../../../../../../src/plugins/kibana_react/common';
+import { OBSERVABILITY_OWNER, SECURITY_SOLUTION_OWNER } from '../../../common';
+import { OWNER_INFO } from '../../../common/constants';
+
+import { FieldHook, getFieldValidityAndErrorMessage, UseField } from '../../common/shared_imports';
+import * as i18n from './translations';
+
+interface MenuSelectionProps {
+  field: FieldHook<string>;
+  isLoading: boolean;
+  availableOwners: string[];
+}
+
+interface Props {
+  availableOwners: string[];
+  isLoading: boolean;
+}
+
+const DEFAULT_SELECTABLE_OWNERS = [SECURITY_SOLUTION_OWNER, OBSERVABILITY_OWNER] as const;
+
+const FIELD_NAME = 'selectedOwner';
+
+const FullWidthKeyPadMenu = euiStyled(EuiKeyPadMenu)`
+  width: 100%;
+`;
+
+const FullWidthKeyPadItem = euiStyled(EuiKeyPadMenuItem)`
+
+  width: 100%;
+`;
+
+const CaseOwnerSelectionComponent: React.FC<Props> = ({ availableOwners, isLoading }) => {
+  return (
+    <UseField
+      path={FIELD_NAME}
+      component={MenuSelection}
+      componentProps={{ availableOwners, isLoading }}
+    />
+  );
+};
+
+const MenuSelection = ({
+  availableOwners,
+  field,
+  isLoading = false,
+}: MenuSelectionProps): JSX.Element => {
+  const { errorMessage, isInvalid } = getFieldValidityAndErrorMessage(field);
+  const radioGroupName = useGeneratedHtmlId({ prefix: 'caseOwnerRadioGroup' });
+
+  const onChange = useCallback((val: string) => field.setValue(val), [field]);
+
+  useEffect(() => {
+    if (!field.value) {
+      onChange(availableOwners[0]);
```

Comment:
Conversation with Pavel and Monina suggested we always have a default selected to help show the user that these are buttons. 

Was curios though, this was a case where I could have something like:
`useEffect( () => {
      onChange(availableOwners[0]);
   }, [ ] )`
   
   In other words 0 dependencies since it only needs to run once. Would that have been preferred or should I leave as is? or doesnt matter?

Reply:
The `availableOwners` array comes from a `Object.entries(capabilities).reduce(...);` call, the order is not guaranteed in the `entries` so the default owner selected may change from one execution to another. I am not sure if this is important for Pavel or Monina, if it is we'll need to define something like a `DEFAULT_SELECTED_OWNER`.

Reply:
Answering your question, I am not sure that the `availableOwners` will be defined (empty array probably) using 0 dependencies. I think it is okay the way it is 👍

Reply:
Thanks for your answer. glad its the right way :D. I spoke to Pavel and security should be the primary default, so Ill go ahead and fix that.